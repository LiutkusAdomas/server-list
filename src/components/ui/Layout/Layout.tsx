import { ReactNode } from 'react';
import { Header } from '../Header/Header';

type LayoutPropsType = {
  children?: ReactNode;
};

export const Layout: React.FC<LayoutPropsType> = ({ children }) => {
  return (
    <>
      <Header />
      <main className="min-h-screen flex justify-center bg-gray-300 p-4 pt-20">{children}</main>
    </>
  );
};
