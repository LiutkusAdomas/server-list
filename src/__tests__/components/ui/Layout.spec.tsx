import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Layout } from '../../../components/ui/Layout/Layout';
import { StaticRouter } from 'react-router-dom/server';

describe('Layout component', () => {
  it('should render the Layout with some text', () => {
    const { asFragment } = render(
      <StaticRouter location="/">
        <Layout>Test</Layout>
      </StaticRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });
});
