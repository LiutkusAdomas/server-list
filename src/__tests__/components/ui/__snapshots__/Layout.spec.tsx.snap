// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout component > should render the Layout with some text 1`] = `[Function]`;
